spark/src/main/scala/org/apache/iotdb/tsfile/DefaultSource.scala

```diff
@@ -137,49 +118,25 @@ private[tsfile] class DefaultSource extends FileFormat with DataSourceRegister {
         // Used to convert `Row`s containing data columns into `InternalRow`s.
         private val encoderForDataColumns = RowEncoder(requiredSchema)
 
-        private var deltaObjectId = "null"
-
         override def hasNext: Boolean = {
-          var hasNext = false
-
-          while (priorityQueue.nonEmpty && !hasNext) {
-            //get a record from priorityQueue
-            val tmpRecord = priorityQueue.dequeue()
-            //insert a record to priorityQueue
-            if (dataSets(tmpRecord.index).hasNextRecord) {
-              priorityQueue.enqueue(Record(dataSets(tmpRecord.index).getNextRecord, tmpRecord.index))
-            }
-
-            if (curRecord == null || tmpRecord.record.timestamp != curRecord.record.timestamp ||
-              !tmpRecord.record.getFields.get(0).deltaObjectId.equals(curRecord.record.getFields.get(0).deltaObjectId)) {
-              curRecord = tmpRecord
-              deltaObjectId = curRecord.record.getFields.get(0).deltaObjectId
-              hasNext = true
-            }
-          }
-
+          val hasNext = queryDataSet.hasNext
           hasNext
         }
 
         override def next(): InternalRow = {
 
+          val curRecord = queryDataSet.next()
           val fields = new scala.collection.mutable.HashMap[String, Field]()
```

Comment:
Check if this map is necessary. The index of a field should be the same as the index of the queried path (skip the time column).

For example:
selected column: series1, series2, time, series3
RowRecord fields: series1Value,  series2Value, series3Value

Then, use a boolean to denote whether the index of a selected path should decrease one to get the value.